scripts/azure/storage-manager.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+set -e
+
+readonly CREATE_ACTION="create"
+readonly DELETE_ACTION="delete"
+
+function usage() {
+    echo "Usage:" 1>&2
+    echo " $0 create" 1>&2
+    echo " $0 delete" 1>&2
+}
+
+case "$1" in
+create)
+    ACTION=$CREATE_ACTION
+    ;;
+delete)
+    ACTION=$DELETE_ACTION
+    ;;
+esac
+shift
+
+function create_resource_group() {
+
+    echo "checking if resource group exists" >&2
+    GROUP_EXISTS=$(az group list --query '[].name' | awk "/${CK8S_ENVIRONMENT_NAME}-storage-resource-group/")
+    if [ "$GROUP_EXISTS" ]; then
+        echo "resource group [${CK8S_ENVIRONMENT_NAME}-storage-resource-group] already exists" >&2
+        echo "contnue using this group ? (y/n)" >&2
+        # shellcheck disable=SC2162
+        read -n 1 cmdinput
+        case "$cmdinput" in
+        y) return ;;
+        n) exit 0 ;;
+        esac
+    else
+        echo "resource group [${CK8S_ENVIRONMENT_NAME}-storage-resource-group] does not exist, creating it now" >&2
+        az group create \
+            --name "$CK8S_ENVIRONMENT_NAME"-storage-resource-group \
+            --location swedencentral --only-show-errors
+    fi
+
+}
+
+function create_storage_account() {
+
+    echo "checking if storage account exists" >&2
+    ACCOUNT_EXISTS=$(az storage account list --query '[].name' | awk "/${CK8S_ENVIRONMENT_NAME}storageaccount/")
+    if [ "$ACCOUNT_EXISTS" ]; then
+        echo "storage account [${CK8S_ENVIRONMENT_NAME}storageaccount] already exists" >&2
```

Comment:
Is there a reason as to why the storage account isn't `${CK8S_ENVIRONMENT_NAME}-storage-account`?
Also do we need the `-<resource-type>` suffixes, isn't it clear from their name what it relates to?
Or do we in the case of the `resource-group` have multiple for a typical Compliant Kubernetes setup?

Reply:
Storage accounts names can only contain lowercase letter and numbers. So even `${CK8S_ENVIRONMENT_NAME}` needs to be in that format.
I just added the suffixes for extra "visibility" it can be removed.